src/editors/containers/ProblemEditor/components/EditProblemView/SettingsWidget/settingsComponents/HintsCard.jsx

```diff
@@ -0,0 +1,46 @@
+import React from 'react'
+import { useDispatch } from 'react-redux';
+import { injectIntl, FormattedMessage, intlShape } from '@edx/frontend-platform/i18n';
+import SettingsOption from '../SettingsOption';
+import { Button } from '@edx/paragon'
+import { Add } from '@edx/paragon/icons';
+import messages from '../messages';
+import { hintsCardHooks, hintsRowHooks } from '../hooks';
+import HintRow from './HintRow';
+
+export const HintsCard = ({
+    hints,
+    intl,
+}) => {
+    const dispatch = useDispatch();
+    const { summary, handleAdd } = hintsCardHooks(hints, dispatch);
```

Comment:
Looking at the general pattern followed throughout the codebase, it seems `mapDispatchToProps` is used to fetch functions which is used directly within the component or are required by a third party imported component. [Example for this type of usage](https://github.com/open-craft/frontend-lib-content-components/blob/0893f3b9a85da20a03e177da1f613ae802d064c9/src/editors/containers/TextEditor/index.jsx#L86-L134).

However, passing of dispatch to hooks is preferred when the dispatch function is only used within the hook. [Example 1](https://github.com/open-craft/frontend-lib-content-components/blob/0893f3b9a85da20a03e177da1f613ae802d064c9/src/editors/containers/EditorContainer/components/TitleHeader/index.jsx#L19-L29), [Example 2](https://github.com/open-craft/frontend-lib-content-components/blob/0893f3b9a85da20a03e177da1f613ae802d064c9/src/editors/Editor.jsx#L19-L29).

Ideally the `useDispatch` method should be invoked within the hook too and not passed from the component, but there apprears to be some limitation to this, as described in the [testability decisions document](https://github.com/edx/frontend-lib-content-components/blob/3b3aa42b28d2601fbb7d24a52b18f2fcd2b6c7ff/docs/decisions/000x-internal-editor-testability-decisions.md#additional-considrations)

I am personally good with using either of the patterns. However, if you know of any advantage of using `mapDispatchToProps` over `useDispatch`, we could perhaps use those arguments to establish a pattern here and justify  it to upstream.

Reply:
>function names like addHintCard and hintSummary makes more sense I feel

These functions are pretty localized within their components/hooks and are _not_ exposed outside their component/hooks and so functionally there is no risk of using `handleAdd` of one component inside another wrong component.

Since, changing these function names for all the different settings components, their hooks as well as their test cases would require some amount of effort, perhaps we make a note of this and comeback to it when we have more bandwith with our budget ?

Reply:
My understanding is very basic here, to keep the visual component as dumb as possible and move the business logic to `hooks`. Hence, moving the function to `mapDispatchToProps` will reduce the dependency on calling a function inside the component and will make the component render depending on the functions passed to the component.

That is the pattern I am willing to keep.

Reply:
@farhaanbukhsh 
Good point. The goal indeed is to eliminate business logic from visual components and move it to hooks.
I agree, calling the `useDispatch` function in the components sort of defeats this goal.

I have therefore removed `useDispatch` call from the different settings components.